### PR TITLE
[doc] Update booster user documents.

### PR DESCRIPTION
**Pull request by Fridge003**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

#4628

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

- Rewrite some docstrings in booster.py
- Update user documents of booster, including booster_api.md, booster_checkpoint.md and booster_plugins.md

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/booster/booster.py [hunk 1]
```diff
@@ -110,14 +112,19 @@ def boost(
...
-            optimizer (Optimizer): The optimizer to be boosted.
-            criterion (Callable): The criterion to be boosted.
-            dataloader (DataLoader): The dataloader to be boosted.
-            lr_scheduler (LRScheduler): The lr_scheduler to be boosted.
+            model (nn.Module): Convert model into a wrapped model for distributive training.
+                               The model might be decorated or partitioned by plugin's strategy after execution of this method.
+            optimizer (Optimizer, optional): Convert optimizer into a wrapped optimizer for distributive training.
+                                             The optimizer's param groups or states might be decorated or partitioned by plugin's strategy after execution of this method. Defaults to None.
+            criterion (Callable, optional): The function that calculates loss. It will not be modified, but will be utilized during wrapping of model and optimizer. Defaults to None.
+            dataloader (DataLoader, optional): The prepared dataloader for training. It will not be modified, but will be utilized during wrapping of model and optimizer. Defaults to None.
```
Comment (ver217) [inline colossalai/booster/booster.py:123]: It may be modified. But we don't modify it now.

Comment (ver217) [reply colossalai/booster/booster.py:123]: So as criterion and lr scheduler

File: docs/source/en/basics/booster_plugins.md [hunk 2]
```diff
@@ -69,4 +68,24 @@ More details can be found in [Pytorch Docs](https://pytorch.org/docs/main/fsdp.h
...
+
+2. Mixed Precision Training: Support for fp16/bf16 mixed precision training. More details about its arguments configuration can be found in [Mixed Precision Training Doc](../features/mixed_precision_training_with_booster.md).
+
+3. Torch DDP: This plugin will automatically adopt Pytorch DDP as data parallel strategy when pipeline parallel and Zero is not used. More details about its arguments configuration can be found in [Pytorch DDP Docs](https://pytorch.org/docs/main/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel).
+
+4. Zero: This plugin can adopt Zero 1/2 as data parallel strategy through setting the `zero_stage` argument as 1 or 2 when initializing plugin. More details about its argument configuration can be found in [Low Level Zero Plugin](#low-level-zero-plugin).
```
Comment (ver217) [inline docs/source/en/basics/booster_plugins.md:82]: Please demonstrate the compatibility of zero and pipeline.

File: colossalai/booster/booster.py (hunk 1 above)
Comment (Fridge003) [reply colossalai/booster/booster.py]: modified

File: docs/source/en/basics/booster_plugins.md (hunk 2 above)
Comment (Fridge003) [reply docs/source/en/basics/booster_plugins.md]: modified

**Review by ver217 — APPROVED** [review]

**Review by FrankLeeeee — APPROVED** [review]
